extractor/src/test/java/com/example/extractor/MainsLauncherTest.java

```diff
@@ -23,7 +29,58 @@ class MainsLauncherTest {
      */
     @Test
     void testMain() throws Exception {
-        // Pass the base path as argument and assert that it doesn't throw an exception
+        // Pass the base path as an argument and assert that it doesn't throw an exception.
         assertDoesNotThrow(() -> CheckstyleExampleExtractor.main(new String[]{CHECKSTYLE_REPO_PATH}));
     }
+
+    /**
+     * Tests the main method with a specific input file and verifies that the output file is generated correctly.
+     *
+     * @param tempDir The temporary directory where the output file will be created.
+     * @throws Exception if any error occurs during the test
+     */
+    @Test
+    void testMainWithInputFile(@TempDir final Path tempDir) throws Exception {
+        // The input file path is declared as final because it is not modified after initialization.
+        final String inputFilePath = "src/test/resources/com.puppycrawl.tools.checkstyle.checks.whitespace.methodparampad.InputFile/InputMethodParamPadWhitespaceAround.java";
+
+        // Create a temporary output file, declared as final since it is not reassigned.
+        final Path outputFile = tempDir.resolve("output-config.xml");
+
+        // Run the main method with the input file and assert that it doesn't throw an exception.
+        assertDoesNotThrow(() -> CheckstyleExampleExtractor.main(new String[]{
+                CHECKSTYLE_REPO_PATH,
+                "--input-file",
+                inputFilePath,
+                outputFile.toString()
+        }));
+
+        // Verify that the output file was created successfully.
+        assertTrue(Files.exists(outputFile), "Output file should be created");
+
+        // Read the content of the output file, declared as final because it is not reassigned.
+        final String generatedContent = Files.readString(outputFile);
+
+        // Verify the content of the generated file, ensuring it includes the expected XML elements.
+        assertThat(generatedContent).contains("<module name=\"MethodParamPad\">");
```

Comment:
Assert comparison of whole file, in your resources should be Input files and exact result of generation.

Reply:
done